src/api/projects/projects.service.ts

```diff
@@ -154,6 +171,80 @@ export class ProjectsService {
           increment: hours,
         },
       },
+      include: {
+        user: true,
+      },
     });
+
+    const totalHours = (
+      await this.prisma.projectUser.aggregate({
+        _sum: {
+          hours: true,
+        },
+        where: {
+          userId,
+        },
+      })
+    )._sum.hours;
+
+    const newBadge = getCorrespondingBadge(totalHours, previousTotalHours);
+
+    if (newBadge) await this.awardNewBadge(newProjectUserRecord.user.email, newBadge);
+
+    return this.prisma.projectUser.findUnique({
+      where: {
+        projectId_userId: {
+          projectId,
+          userId,
+        },
+      },
+    });
+  }
+
+  public async awardNewBadge(email: string, newBadge: string) {
```

Comment:
What do you think in moving all the logic from the badging feature to another file called: `badgr.service.ts`?